packages/core/framework/src/config/types.ts

```diff
@@ -1,978 +1,8 @@
-import {
-  ExternalModuleDeclaration,
-  InternalModuleDeclaration,
+export {
+  AdminOptions,
+  SessionOptions,
+  ProjectConfigOptions,
+  HttpCompressionOptions,
+  ConfigModule,
+  PluginDetails,
 } from "@medusajs/types"
```

Comment:
These were duplicated with slight variations. Ideally we should have one copy. But happy to revert if there was a reason for duplicating them

Reply:
I guess this is technically a breaking change, so should we instead deprecate the types? We can perhaps assign these types to the other exported ones to remove the duplicated code.

Reply:
I think renaming these to something else will possibly create two copies of types. One exported from `@medusajs/types` and one exported from `@medusajs/framework` with a different name.

So, if we want to prevent the breaking change, then I can revert this

Reply:
Do we have to rename them?

The thinking here would be to deprecate them and do:
```
import { TypeFromTypesPackage } from "@medusajs/types"

export type ThisType = TypeFromTypesPackage
```

Reply:
Actually the names were the same. Now, if I want to deprecate the one from the framework package then either they have to named differently or the re-export from the types package has to have a different name.

```ts
import { AdminOptions } from '@medusajs/types'
export { AdminOptions }

// 👇 Duplicate name
// @deprecated
export type AdminOptions = {} 
```